controlplane/src/core/repositories/SubgraphRepository.ts

```diff
@@ -874,6 +875,11 @@ export class SubgraphRepository {
       conditions.push(sql.raw(`labels && '{${labelsSQL}}'`));
     }
 
+    // Only get subgraphs that do not have any labels if the label matchers are empty.
+    if (data.labelMatchers.length === 0) {
+      conditions.push(eq(targets.labels, []));
+    }
+
     const subgraphs = await this.db
```

Comment:
We forget to use `tx` here.

Reply:
The repository is created in a transaction, all good here

Reply:
ok